lib/server.js

```diff
@@ -44,196 +39,200 @@ module.exports = Server
-}
+  /**
+   * Helper function to subscribe to all queues.
+   * @private
+   * @return {promise} Resolved when queues are all subscribed.
+   */
+  _subscribeAll () {
+    debug('_subscribeAll')
+    return Promise.resolve(this.hermes.getQueues()).bind(this)
+      .map(this._subscribe)
+  }
 
-/**
- * Helper function to unsubscribe from all queues.
- * @private
- * @return {promise} Resolved when queues are all subscribed.
- */
-Server.prototype._unsubscribeAll = function () {
-  debug('_unsubscribeAll')
-  return Promise.resolve(this.hermes.getQueues()).bind(this)
-    .map(this._unsubscribe)
-}
+  /**
+   * Helper function that unsubscribes to a given queue name.
+   * @private
+   * @param {string} queueName Name of the queue.
+   * @return {promise} A promise that resolves post worker creation
+   *   and subscription.
+   */
+  _unsubscribe (queueName) {
+    debug('_unsubscribe(' + queueName + ')')
+    this.log.trace('ponos.Server: unsubscribing from ' + queueName)
+    const handler = null // null will remove all handlers
+    return this.hermes.unsubscribeAsync(queueName, handler)
+  }
 
-/**
- * Runs a worker for the given queue name, job, and acknowledgement callback.
- * @private
- * @param {string} queueName Name of the queue.
- * @param {object} job Job for the worker to perform.
- * @param {function} done RabbitMQ acknowledgement callback.
- */
-Server.prototype._runWorker = function (queueName, job, done) {
-  debug('_runWorker(' + [ queueName, job ].join(', ') + ')')
-  var opts = clone(this._workerOptions[queueName])
-  defaults(opts, {
-    queue: queueName,
-    job: job,
-    task: this._tasks[queueName],
-    done: done,
-    log: this.log,
-    errorCat: this.errorCat
-  })
-  Worker.create(opts)
-}
+  /**
+   * Helper function to unsubscribe from all queues.
+   * @private
+   * @return {promise} Resolved when queues are all subscribed.
+   */
+  _unsubscribeAll () {
+    debug('_unsubscribeAll')
+    return Promise.resolve(this.hermes.getQueues()).bind(this)
+      .map(this._unsubscribe)
+  }
 
-/**
- * Starts the worker server and listens for jobs coming from all queues.
- * @return {promise} A promise that resolves when the server is listening.
- */
-Server.prototype.start = function () {
-  debug('start')
-  this.log.trace('ponos.Server: starting')
-  return this.hermes.connectAsync().bind(this)
-    .then(this._subscribeAll)
-    .then(function () {
-      this.log.trace('ponos.Server: started')
-    })
-    .catch(function (err) {
-      this.errorCat.report(err)
-      throw err
+  /**
+   * Runs a worker for the given queue name, job, and acknowledgement callback.
+   * @private
+   * @param {string} queueName Name of the queue.
+   * @param {object} job Job for the worker to perform.
+   * @param {function} done RabbitMQ acknowledgement callback.
+   */
+  _runWorker (queueName, job, done) {
+    debug('_runWorker(' + [ queueName, job ].join(', ') + ')')
```

Comment:
This is  great example of the aforementioned usefulness of debug. With a `log.trace` we could simply add the queue name and job as data to the log line. This would get rid of the "wonky" `[thing, thing2, thing3].join(', ')` stuff you have to do in order for it to fit the debug model.

Also having two different types of console output can be really confusing so you often only use one anyway.

Furthermore it reduces the syntactic complexity of programs overall if there is less garbage and more substance to each of the functions.